evalml/utils/nullable_type_utils.py

```diff
@@ -0,0 +1,103 @@
+import woodwork as ww
+
+
+def _downcast_nullable_X(X, handle_boolean_nullable=True, handle_integer_nullable=True):
+    """Removes Pandas nullable integer and nullable boolean dtypes from data by transforming
+        to other dtypes via Woodwork logical type transformations.
+
+    Args:
+        X (pd.DataFrame): Input data of shape [n_samples, n_features] whose nullable types will be changed.
+        handle_boolean_nullable (bool, optional): Whether or not to downcast data with BooleanNullable logical types.
+        handle_integer_nullable (bool, optional): Whether or not to downcast data with IntegerNullable or AgeNullable logical types.
+
+
+    Returns:
+        X with any incompatible nullable types downcasted to compatible equivalents.
+    """
+    if X.ww.schema is None:
+        X.ww.init()
+
+    select_param = []
+    if handle_boolean_nullable:
+        select_param.append("BooleanNullable")
+    if handle_integer_nullable:
+        select_param.append("IntegerNullable")
+        select_param.append("AgeNullable")
+
+    if not select_param:
+        return X
```

Comment:
I think this check is redundant, we can check for the same scenario with `X.ww.select([])` below, right?

Reply:
Correct--the benefit is just that we wouldn't have to do `select` since we'd already know there was nothing problematic. This is me maybe over engineering a bit here for performance, thinking about a case where we have a very wide dataset that we'd be happy to avoid the `select` call on.

But I think in reality, we're not going to be calling `_downcast_nullable_X` on data that has no nullable type incompatibilities. This is because of a change I made in `ComponentBase._handle_nullable_types`; originally I had us calling the downcast function on `X` every time even if it had no incompatibilities, but I changed it and forgot to remove this logic. Will remove!

Reply:
Removed: 1ad54d33c76c681af0e187d7dbca5f3066c63402